feat(napi/parser): add `convert_span_utf16` option

hi-ogawa opened the pull request:
For starter, I added opt-in `convert_span_utf16` for `napi/parser` level option, so rolldown-vite can start to battle test estree output.

hi-ogawa commented:
* **#8983** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8983?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8983?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8983).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (commented):
Eventually this should be an `estree` option. With proper documentation regarding its performance characteristics.

At present we should document it as for internal use only.

Boshen commented:
### Merge activity

* **Feb 11, 1:21 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Feb 11, 1:21 AM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

hi-ogawa committed:
feat(napi/parser): add `convert_span_utf16` option (#8983)

For starter, I added opt-in `convert_span_utf16` for `napi/parser` level option, so rolldown-vite can start to battle test estree output.
